paymentsheet/src/main/java/com/stripe/android/customersheet/StripeCustomerAdapter.kt

```diff
@@ -33,75 +35,87 @@ internal class StripeCustomerAdapter @Inject constructor(
     private var cachedCustomerEphemeralKey: CachedCustomerEphemeralKey? = null
 
     override suspend fun retrievePaymentMethods(): Result<List<PaymentMethod>> {
-        return getCustomerEphemeralKey().fold(
-            onSuccess = { customer ->
-                val paymentMethods = customerRepository.getPaymentMethods(
+        return withContext(Dispatchers.IO) {
```

Comment:
Instead of wrapping everything here, can we push it down further? It looks like `customerRepository.getPaymentMethods()` already uses a `workContext`, so you’d only need to wrap `getCustomerEphemeralKey()` in an IO context.